cetmix_tower_server/security/cx_tower_plan_line_action_security.xml

```diff
@@ -6,18 +6,54 @@
         <field name="name">Tower plan line action: user access rule</field>
         <field name="model_id" ref="model_cx_tower_plan_line_action" />
         <field name="groups" eval="[(4, ref('cetmix_tower_server.group_user'))]" />
-        <field name="domain_force">[('access_level', '=', '1')]</field>
+        <field name="domain_force">
+            [
+            ('access_level', '=', '1'),
+            '|',
+            ('line_id.plan_id.server_ids', '=', False),
+            ('line_id.plan_id.server_ids.message_partner_ids', 'in', [user.partner_id.id])
+
+            ]
+        </field>
 
 
     </record>
 
     <record id="cx_tower_plan_line_action_rule_group_manager_access" model="ir.rule">
-        <field name="name">Tower plan line action: manager access rule</field>
+        <field
+            name="name"
+        >Tower plan line action: manager Read/Write access rule</field>
         <field name="model_id" ref="model_cx_tower_plan_line_action" />
-        <field name="domain_force">[('access_level', '&lt;=', '2')]</field>
+        <field name="domain_force">
+            [
+            ('access_level', '&lt;=', '2'),
+            '|',
+            ('line_id.plan_id.server_ids', '=', False),
+            ('line_id.plan_id.server_ids.message_partner_ids', 'in', [user.partner_id.id])
+            ]
+        </field>
         <field name="groups" eval="[(4, ref('cetmix_tower_server.group_manager'))]" />
+        <field name="perm_read" eval="1" />
+        <field name="perm_create" eval="1" />
+        <field name="perm_write" eval="1" />
+        <field name="perm_unlink" eval="0" />
 
+    </record>
 
+    <record id="cx_tower_plan_line_action_rule_manager_unlink" model="ir.rule">
+        <field name="name">Tower plan line action: manager delete own records</field>
+        <field name="model_id" ref="model_cx_tower_plan_line_action" />
+        <field name="domain_force">
+            [
+            ('create_uid', '=', user.id),
+            ('access_level', '&lt;=', '2'),
+            '|',
+            ('line_id.plan_id.server_ids', '=', False),
+            ('line_id.plan_id.server_ids.message_partner_ids', 'in', [user.partner_id.id])
+            ]
+        </field>
+        <field name="groups" eval="[(4, ref('cetmix_tower_server.group_manager'))]" />
+        <field name="perm_unlink" eval="1" />
```

Comment:
"1" is the default value for the "perm_..." fields. Which means you need to set it to "0" explicitly. Otherwise this rule is applied to all other actions too.
NB: this case should have been caught by tests, looks like they should be fixed or updated.

Reply:
Thank you for noticing, you are right, as in this case manager should have full access to the plan_line_action he has created, I don't need set this value to "1" expicitly, as this is default.

Test cases related to the manager's ability to delete plan_line_actions are covered [here](https://github.com/cetmix/cetmix-tower/blob/75c5fb668cf66618af4bd5886f43451fdfac5c85/cetmix_tower_server/tests/test_plan.py#L1483) and below, if you think it is insufficient please let me know.